common.proto

```diff
@@ -0,0 +1,78 @@
+syntax = "proto3";
+
+option go_package = ".;pb";
+
+package common;
+
+// Identity is included in every message that requires auth. One AuthMethod at a
+// time is supported. For now, the only supported method is a token. These need
+// to be created manually and distributed to any clients.
+message Identity {
+  oneof AuthMethod {
+    string token = 1;
+  }
+}
+
+// When sent by a chat backend, the ID MAY optionally be used as a hint to the
+// frontend to get the same external ID on reconnection. When sent by core to a
+// plugin, the ID MUST be an ID pointing to an exact backend instance.
+message Backend {
+  string id = 1;
+  string type = 2;
+}
+
+// When sent by a chat backend, the ID MUST be a unique ID representing this
+// user in the given chat service. When sent by core to a plugin, the ID MUST be
+// an ID pointing to an instance of a user on a service.
+message User {
+  string id = 1;
+  string display_name = 2;
+}
+
+// When sent by a chat backend, the ID MUST be a unique ID representing this
+// channel in the given chat service. When sent by core to a plugin, the ID MUST
+// be an ID pointing to an instance of a channel on a service.
+message Channel {
+  string id = 1;
+  string display_name = 2;
+  string topic = 3;
+}
+
+// ChannelSource represents where a message came from. Note that User is used
+// rather than just an ID so any clients can have access to the name as well.
+// This may change in the future.
+message ChannelSource {
+  User user = 1;
+  string channel_id = 2;
```

Comment:
nit: in my head these should be reversed because you're saying this is from a channel, and that's the second field lol